### ISSUE TITLE
Clarify documentation for sentinel behavior of `*encode*` fns

### ISSUE THREAD

**Issue by starthal** [jamesmunns/cobs.rs#18]
The documentation describes that `encode`, `try_encode`, etc. "use" the default sentinel value (or override it). IMO they should clarify that the functions do not actually write the sentinel to the buffer.

This is documented in `CobsEncoder::finalize` but not the free functions that use it.

**Comment by rockboynton** [comment]
I agree, I would really expect `encode` to append the sentinel as is the point of COBS

**Comment by robamu** [comment]
This is definitely a good idea. I'll try to open a PR with the suggested improvement

**Comment by robamu** [comment]
See #29

**Comment by starthal** [comment]
Thanks!
